Hardware/CPU/AMD17CPU.cs

```diff
@@ -172,320 +173,325 @@ public void UpdateSensors()
+        _hw.ActivateSensor(_power);
+        _hw.ActivateSensor(_vcore);
+      }
+
+      #region UpdateSensors
+      public void UpdateSensors()
+      {
+        // CPUID cpu = threads.FirstOrDefault();
+        CPUID cpu = Threads[0];
+        if (cpu == null)
+          return;
+        uint eax, edx;
+        ulong mask = Ring0.ThreadAffinitySet(1UL << cpu.Thread);
+
+        // MSRC001_0299
+        // TU [19:16]
+        // ESU [12:8] -> Unit 15.3 micro Joule per increment
+        // PU [3:0]
+        Ring0.Rdmsr(MSR_PWR_UNIT, out eax, out edx);
+        int tu = (int)((eax >> 16) & 0xf);
+        int esu = (int)((eax >> 12) & 0xf);
+        int pu = (int)(eax & 0xf);
+
+        // MSRC001_029A
+        // total_energy [31:0]
+        DateTime sample_time = DateTime.Now;
+        Ring0.Rdmsr(MSR_CORE_ENERGY_STAT, out eax, out edx);
+        uint total_energy = eax;
+
+        // MSRC001_0293
+        // CurHwPstate [24:22]
+        // CurCpuVid [21:14]
+        // CurCpuDfsId [13:8]
+        // CurCpuFid [7:0]
+        Ring0.Rdmsr(MSR_HARDWARE_PSTATE_STATUS, out eax, out edx);
+        int CurHwPstate = (int)((eax >> 22) & 0x3);
+        int CurCpuVid = (int)((eax >> 14) & 0xff);
+        int CurCpuDfsId = (int)((eax >> 8) & 0x3f);
+        int CurCpuFid = (int)(eax & 0xff);
+
+        // MSRC001_0064 + x
+        // IddDiv [31:30]
+        // IddValue [29:22]
+        // CpuVid [21:14]
+        // CpuDfsId [13:8]
+        // CpuFid [7:0]
+        // Ring0.Rdmsr(MSR_PSTATE_0 + (uint)CurHwPstate, out eax, out edx);
+        // int IddDiv = (int)((eax >> 30) & 0x03);
+        // int IddValue = (int)((eax >> 22) & 0xff);
+        // int CpuVid = (int)((eax >> 14) & 0xff);
+        Ring0.ThreadAffinitySet(mask);
+
+        // clock
+        // CoreCOF is (Core::X86::Msr::PStateDef[CpuFid[7:0]] / Core::X86::Msr::PStateDef[CpuDfsId]) * 200
+        _clock.Value = (float)((double)CurCpuFid / (double)CurCpuDfsId * 200.0);
+
+        // multiplier
+        _multiplier.Value = (float)((double)CurCpuFid / (double)CurCpuDfsId * 2.0);
+
+        // Voltage
+        double VIDStep = 0.00625;
+        double vcc = 1.550 - (double)VIDStep * CurCpuVid;
+        _vcore.Value = (float)vcc;
+
+        // power consumption
+        // power.Value = (float) ((double)pu * 0.125);
+        // esu = 15.3 micro Joule per increment
+        if (_lastPwrTime.Ticks == 0)
+        {
+          _lastPwrTime = sample_time;
+          _lastPwrValue = total_energy;
+        }
+        // ticks diff
+        TimeSpan time = sample_time - _lastPwrTime;
+        long pwr;
+        if (_lastPwrValue <= total_energy)
+          pwr = total_energy - _lastPwrValue;
+        else
+          pwr = (0xffffffff - _lastPwrValue) + total_energy;
+
+        // update for next sample
+        _lastPwrTime = sample_time;
+        _lastPwrValue = total_energy;
+
+        double energy = 15.3e-6 * pwr;
+        energy /= time.TotalSeconds;
+
+        if(!double.IsNaN(energy))

```

Comment:
formatting

Reply:
What is wron with the formatting?
i use a . editorconfig. the code style of the project is terrible to read.
the deafualt microsoft style is murch more readable.

`
# 
# Documentation: http://EditorConfig.org
# VS15 plug-in: http://visualstudiogallery.msdn.microsoft.com/c8bccfe2-650c-4b42-bc5c-845e21f96328
# Natively supported in VS17
#
# Configure editor for this project to match original project indentation scheme.

# top-most EditorConfig file
root = true

[*]
trim_trailing_whitespace = true

[*.{cs,vb}]
indent_style = space
indent_size = 2
dotnet_style_object_initializer = true:suggestion
csharp_style_pattern_matching_over_is_with_cast_check = true:suggestion



# New line preferences
csharp_new_line_before_open_brace = none
csharp_new_line_before_else = false
csharp_new_line_before_catch = false
csharp_new_line_before_finally = false
csharp_new_line_before_members_in_object_initializers = true
csharp_new_line_before_members_in_anonymous_types = true
csharp_new_line_between_query_expression_clauses = true
`

Reply:
No space before if. I agree, formatting is terrible, it's still original from OHM in order to keep PRs as easy as possible.

Reply:
i open a new issue to discuss the future development.